lib/evm/eei.ts

```diff
@@ -90,17 +88,17 @@ export default class EEI {
    * @param amount - Amount of gas refunded
    */
   refundGas(amount: BN): void {
-    this._result.gasRefund.iadd(amount)
+    this._evm._refund.iadd(amount)
   }
 
   /**
    * Reduces amount of gas to be refunded by a positive value.
    * @param amount - Amount to subtract from gas refunds
    */
   subRefund(amount: BN): void {
-    this._result.gasRefund.isub(amount)
-    if (this._result.gasRefund.ltn(0)) {
-      this._result.gasRefund = new BN(0)
+    this._evm._refund.isub(amount)
+    if (this._evm._refund.ltn(0)) {
+      this._evm._refund = new BN(0)
       trap(ERROR.REFUND_EXHAUSTED)
     }
```

Comment:
Maybe we can do `this._result.gasRefund = this._evm._refund` after modifying it. It clutters the code a little, but it avoids a breaking change. 

As David, said Ganache only depends on `gasRefund` at the end of every tx, so this would work.

Reply:
I re-added `gasRefund` to `ExecResult`, by copying `EVM._refund` at the end of message execution in EVM. This should work for the Ganache use-case. I think we should be backwards-compatible now, at least for the most part.